linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/rule.yml

```diff
@@ -0,0 +1,31 @@
+documentation_complete: true
+
+prodtype: fedora
+
+title: 'Configure OpenSSL library to use System Crypto Policy'
+
+description: |-
+    Crypto Policies provide a centralized control over crypto algorithms usage of many packages.
+    OpenSSL is supported by crypto policy, but the OpenSSL configuration may be
+    set up to ignore it.
+    To check that Crypto Policies settings are configured correctly, you have to examine the OpenSSL config file
+    available under <tt>/etc/pki/tls/openssl.cnf</tt>.
+    This file has the <tt>ini</tt> format, and it enables crypto policy support
+    if there is a <tt>[ crypto_policy ]</tt> section that contains the <tt>.include /etc/crypto-policies/back-ends/openssl.config</tt> directive.
+
+rationale: |-
+    Overriding the system crypto policy makes the behavior of the Java runtime violates expectations,
+    and makes system configuration more fragmented.
+
+severity: unknown
+
+ocil_clause: |-
+    the OpenSSL config file doesn't contain the whole section,
+    or that the section doesn't have the <pre>.include /etc/crypto-policies/back-ends/openssl.config</pre> directive
+
+ocil: |-
+    To verify that OpenSSL uses the system crypro policy, check out that the OpenSSL config file
+    <pre>/etc/pki/tls/openssl.cnf</pre> contains the <pre>[ crypto_policy ]</pre> section with the
+    <pre>.include /etc/crypto-policies/back-ends/openssl.config</pre> directive:
+    <pre>grep -A 3 '\[\s*crypto_policy\s*\]\s*$' /etc/pki/tls/openssl.cnf | grep '^\s*\.include \s*/etc/crypto-policies/back-ends/openssl.config\s*$' </pre>.
```

Comment:
@matejak is there a way to make this grep easier? It's really cumbersome which makes it harder for auditors.

Reply:
I can only make it more complicated :-( It is not technically possible to ensure that an `.ini` file has correct syntax just with grep.

Reply:
@matejak let's make the grep the following:

`grep -A3 crypto_policy /etc/pki/tls/openssl.cnf`

Since OCIL is used for audits, we don't really need to worry about handling the syntax, and it's much easier to deal with.

Reply:
OK, I went for a simplification that checks that the policies config file is at least mentioned in the right context, but it doesn't guarantee at all that it is present in the right section :-(